Automattic/studio
# Studio: Add ability to connect and disconnect a site

Closes: https://github.com/Automattic/dotcom-forge/issues/9354
Closes: https://github.com/Automattic/dotcom-forge/issues/9357

## Proposed Changes

This PR adds the ability to connect and disconnect a site and ensures that the data is persistent per user.  In this PR, we are moving away from relying on the local state and instead save the data about the connected sites to the `appdata-v1.json`.

## Testing Instructions

**Testing per user**

* Log into Studio with account for user A 
* Navigate to `Sync` tab on the site A 
* Click on `Connect site`
* Observe that you see a site connected 
* Switch to the site B
* Observe that you see no connected site on the site B
* Go back to site A to the `Sync` tab
* Click on `Connect site`
* Select a site that both has a staging site and a production site
* Connect that site
* Observe that the two sites get added together
* Click on `Disconnect` button
* Confirm that you see the warning about disconnecting the site 
* Confirm that the warning contains `Don't ask again` option
* Disconnect the site
* Observe that the two sites get disconnected together
* Generally switch between tabs and sites and confirm that the data persists

----
* Log out and log back in with user B
* Observe that you see no sites connected for user B (as expected)
* Connect some sites to user B
* Log out
* Log in with user A
* Confirm that no connected sites from user B are present for user A 

## Pre-merge Checklist
